product_multi_company/tests/test_product_multi_company.py

```diff
@@ -11,8 +11,7 @@
 class TestProductMultiCompany(ProductMultiCompanyCommon, common.TransactionCase):
     def test_create_product(self):
         product = self.env["product.product"].create({"name": "Test"})
-        company = self.env.company
-        self.assertTrue(company.id in product.company_ids.ids)
+        self.assertFalse(product.company_ids)
```

Comment:
issue: wrong test fix.

This test was checking that the current company was added to `product.company_ids`. That's exactly the purpose of this module.

If you commit this change and the tests go ✔️, it means that something's not working properly in the module. Instead, probably you should undo this change and actually fix the module.

Reply:
@yajo 
We will check on it.
Have you had a chance to look into this comments?
https://github.com/OCA/multi-company/pull/459#discussion_r1365327722
https://github.com/OCA/multi-company/pull/459#discussion_r1365388411

Please review and let us know your feedback, that will be very helpful.

Thanks

Reply:
> issue: wrong test fix.
> 
> This test was checking that the current company was added to `product.company_ids`. That's exactly the purpose of this module.
> 
> If you commit this change and the tests go ✔️, it means that something's not working properly in the module. Instead, probably you should undo this change and actually fix the module.

We have removed the default value from the company_ids fields.
Please review this link
https://github.com/OCA/multi-company/pull/458/files#diff-768c66d1106f5fdcb827fc104212e9b76a2739fae84f770ffe2687ec3bea1c4bR19

And this is to support the statement that if a product doesn't have any companies set then it is shared by all the company.

I think we just need to rename the test method name. Correct me If I misunderstood something.

Thanks